I can not sync project after I tried to add your lint rules to the proejct

**Describe the bug**
Hi people, super thank you for sharing your lint rules. I found them valuable. However, I can not sync project in the Android Studio now. However, I can run lint checks from the command line well.

**To Reproduce**
Steps to reproduce the behavior:
1. Add uber lint rules to the project with AGP 4.1.1 and Gradle 6.7.1
2. Try to sync project
3. See error that project can not be sync

**Expected behavior**
I can sync the project

**Error log**
A problem occurred configuring project ':presentation'.
> Failed to notify project evaluation listener.
   > Cannot change attributes of dependency configuration ':presentation:variantReleaseUnitTestCompileClasspath' after it has been resolved
   > KotlinJvmAndroidCompilation with name 'variantReleaseUnitTest' not found.

2020-11-14 10:11:32,074 [6961108]   INFO - .project.GradleProjectResolver - Gradle project resolve error 
org.gradle.tooling.GradleConnectionException: Could not run phased build action using Gradle distribution 'https://services.gradle.org/distributions/gradle-6.7-bin.zip'.
	at org.gradle.tooling.internal.consumer.ExceptionTransformer.transform(ExceptionTransformer.java:55)
	at org.gradle.tooling.internal.consumer.ExceptionTransformer.transform(ExceptionTransformer.java:29)
	at org.gradle.tooling.internal.consumer.ResultHandlerAdapter.onFailure(ResultHandlerAdapter.java:43)
	at org.gradle.tooling.internal.consumer.async.DefaultAsyncConsumerActionExecutor.lambda$run$0(DefaultAsyncConsumerActionExecutor.java:57)
	at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:64)
	at org.gradle.internal.concurrent.ManagedExecutorImpl$1.run(ManagedExecutorImpl.java:48)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at org.gradle.internal.concurrent.ThreadFactoryImpl$ManagedThreadRunnable.run(ThreadFactoryImpl.java:56)
	at java.lang.Thread.run(Thread.java:748)
Caused by: org.gradle.api.UncheckedIOException: java.io.StreamCorruptedException: invalid type code: 39
	at org.gradle.internal.UncheckedException.throwAsUncheckedException(UncheckedException.java:61)
	at org.gradle.internal.UncheckedException.throwAsUncheckedException(UncheckedException.java:41)
	at org.gradle.tooling.internal.provider.serialization.PayloadSerializer.deserialize(PayloadSerializer.java:78)
	at org.gradle.tooling.internal.provider.ProviderConnection.throwFailure(ProviderConnection.java:231)
	at org.gradle.tooling.internal.provider.ProviderConnection.run(ProviderConnection.java:219)
	at org.gradle.tooling.internal.provider.ProviderConnection.runPhasedAction(ProviderConnection.java:179)
	at org.gradle.tooling.internal.provider.DefaultConnection.run(DefaultConnection.java:245)
	at org.gradle.tooling.internal.consumer.connection.PhasedActionAwareConsumerConnection.run(PhasedActionAwareConsumerConnection.java:56)
	at org.gradle.tooling.internal.consumer.connection.ParameterValidatingConsumerConnection.run(ParameterValidatingConsumerConnection.java:62)
	at org.gradle.tooling.internal.consumer.DefaultPhasedBuildActionExecuter$1.run(DefaultPhasedBuildActionExecuter.java:78)
	at org.gradle.tooling.internal.consumer.DefaultPhasedBuildActionExecuter$1.run(DefaultPhasedBuildActionExecuter.java:70)
	at org.gradle.tooling.internal.consumer.connection.LazyConsumerActionExecutor.run(LazyConsumerActionExecutor.java:87)
	at org.gradle.tooling.internal.consumer.connection.CancellableConsumerActionExecutor.run(CancellableConsumerActionExecutor.java:45)
	at org.gradle.tooling.internal.consumer.connection.ProgressLoggingConsumerActionExecutor.run(ProgressLoggingConsumerActionExecutor.java:61)
	at org.gradle.tooling.internal.consumer.connection.RethrowingErrorsConsumerActionExecutor.run(RethrowingErrorsConsumerActionExecutor.java:38)
	at org.gradle.tooling.internal.consumer.async.DefaultAsyncConsumerActionExecutor.lambda$run$0(DefaultAsyncConsumerActionExecutor.java:55)
	... 6 more
Caused by: java.io.StreamCorruptedException: invalid type code: 39
	at java.io.ObjectInputStream.readObject0(ObjectInputStream.java:1656)
	at java.io.ObjectInputStream.defaultReadFields(ObjectInputStream.java:2342)
	at java.io.ObjectInputStream.readSerialData(ObjectInputStream.java:2266)
	at java.io.ObjectInputStream.readOrdinaryObject(ObjectInputStream.java:2124)
	at java.io.ObjectInputStream.readObject0(ObjectInputStream.java:1625)
	at java.io.ObjectInputStream.readArray(ObjectInputStream.java:2030)
	at java.io.ObjectInputStream.readObject0(ObjectInputStream.java:1613)
	at java.io.ObjectInputStream.defaultReadFields(ObjectInputStream.java:2342)
	at java.io.ObjectInputStream.readSerialData(ObjectInputStream.java:2266)
	at java.io.ObjectInputStream.readOrdinaryObject(ObjectInputStream.java:2124)
	at java.io.ObjectInputStream.readObject0(ObjectInputStream.java:1625)
	at java.io.ObjectInputStream.readObject(ObjectInputStream.java:465)
	at java.io.ObjectInputStream.readObject(ObjectInputStream.java:423)
	at java.util.ArrayList.readObject(ArrayList.java:797)
	at sun.reflect.GeneratedMethodAccessor668.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at java.io.ObjectStreamClass.invokeReadObject(ObjectStreamClass.java:1170)
	at java.io.ObjectInputStream.readSerialData(ObjectInputStream.java:2233)
	at java.io.ObjectInputStream.readOrdinaryObject(ObjectInputStream.java:2124)
	at java.io.ObjectInputStream.readObject0(ObjectInputStream.java:1625)
	at java.io.ObjectInputStream.defaultReadFields(ObjectInputStream.java:2342)
	at java.io.ObjectInputStream.readSerialData(ObjectInputStream.java:2266)
	at java.io.ObjectInputStream.readOrdinaryObject(ObjectInputStream.java:2124)
	at java.io.ObjectInputStream.readObject0(ObjectInputStream.java:1625)
	at java.io.ObjectInputStream.defaultReadFields(ObjectInputStream.java:2342)
	at java.io.ObjectInputStream.readSerialData(ObjectInputStream.java:2266)
	at java.io.ObjectInputStream.readOrdinaryObject(ObjectInputStream.java:2124)
	at java.io.ObjectInputStream.readObject0(ObjectInputStream.java:1625)
	at java.io.ObjectInputStream.defaultReadFields(ObjectInputStream.java:2342)
	at java.io.ObjectInputStream.readSerialData(ObjectInputStream.java:2266)
	at java.io.ObjectInputStream.readOrdinaryObject(ObjectInputStream.java:2124)
	at java.io.ObjectInputStream.readObject0(ObjectInputStream.java:1625)
	at java.io.ObjectInputStream.defaultReadFields(ObjectInputStream.java:2342)
	at java.io.ObjectInputStream.readSerialData(ObjectInputStream.java:2266)
	at java.io.ObjectInputStream.readOrdinaryObject(ObjectInputStream.java:2124)
	at java.io.ObjectInputStream.readObject0(ObjectInputStream.java:1625)
	at java.io.ObjectInputStream.defaultReadFields(ObjectInputStream.java:2342)
	at java.io.ObjectInputStream.readSerialData(ObjectInputStream.java:2266)
	at java.io.ObjectInputStream.readOrdinaryObject(ObjectInputStream.java:2124)
	at java.io.ObjectInputStream.readObject0(ObjectInputStream.java:1625)
	at java.io.ObjectInputStream.readObject(ObjectInputStream.java:465)
	at java.io.ObjectInputStream.readObject(ObjectInputStream.java:423)
	at org.gradle.tooling.internal.provider.serialization.PayloadSerializer.deserialize(PayloadSerializer.java:76)
	... 19 more

--- a/build.gradle
+++ b/build.gradle
@@ -43,7 +43,7 @@ subprojects {
     }
 
     apply from: rootProject.file('gradle/dependencies.gradle')
-    apply plugin: 'com.diffplug.gradle.spotless'
+    apply plugin: 'com.diffplug.spotless'
     apply from: rootProject.file('gradle/spotless.gradle')
 
     apply plugin: 'net.ltgt.errorprone'

--- a/gradle/dependencies.gradle
+++ b/gradle/dependencies.gradle
@@ -15,7 +15,7 @@
  */
 
 def versions = [
-        kotlin: '1.4.10',
+        kotlin: '1.4.20',
         support: '27.1.1',
         lint: '27.1.1',
         errorProne      : '2.3.3',
@@ -34,7 +34,6 @@ def apt = [
 ]
 
 def build = [
-        androidPlugin       : "com.android.tools.build:gradle:4.1.1",
         compileSdkVersion   : 28,
         ci: 'true' == System.getenv('CI'),
         minSdkVersion       : 19,
@@ -47,7 +46,7 @@ def build = [
         nullAway             : 'com.uber.nullaway:nullaway:0.7.0',
         gradlePlugins : [
                 android   : '4.1.1',
-                kotlin    : '1.4.10',
+                kotlin    : "${versions.kotlin}",
                 errorProne: '0.8',
                 nullAway  : '0.2',
                 spotless  : '5.8.2',

--- a/gradle/dependencies.gradle
+++ b/gradle/dependencies.gradle
@@ -15,9 +15,9 @@
  */
 
 def versions = [
-        kotlin: '1.3.41',
+        kotlin: '1.4.10',
         support: '27.1.1',
-        lint: '26.5.3',
+        lint: '27.1.1',
         errorProne      : '2.3.3',
         gjf             : '1.7',
         ktlint: '0.31.0'
@@ -34,8 +34,7 @@ def apt = [
 ]
 
 def build = [
-        androidPlugin       : "com.android.tools.build:gradle:3.5.0",
-        buildToolsVersion   : "28.0.3",
+        androidPlugin       : "com.android.tools.build:gradle:4.1.1",
         compileSdkVersion   : 28,
         ci: 'true' == System.getenv('CI'),
         minSdkVersion       : 19,
@@ -47,11 +46,11 @@ def build = [
         errorProneTestHelpers: "com.google.errorprone:error_prone_test_helpers:${versions.errorProne}",
         nullAway             : 'com.uber.nullaway:nullaway:0.7.0',
         gradlePlugins : [
-                android   : '3.5.0',
-                kotlin    : '1.3.31',
+                android   : '4.1.1',
+                kotlin    : '1.4.10',
                 errorProne: '0.8',
                 nullAway  : '0.2',
-                spotless  : '3.23.0',
+                spotless  : '5.8.2',
         ],
         lint: [
             api: "com.android.tools.lint:lint-api:${versions.lint}",

--- a/gradle/wrapper/gradle-wrapper.properties
+++ b/gradle/wrapper/gradle-wrapper.properties
@@ -3,4 +3,4 @@ distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-5.4.1-all.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-6.7.1-all.zip

--- a/lint-checks-rxjava/build.gradle
+++ b/lint-checks-rxjava/build.gradle
@@ -15,7 +15,6 @@
  */
 
 plugins {
-    id 'java-library'
     id 'org.jetbrains.kotlin.jvm'
     id 'org.jetbrains.kotlin.kapt'
     id 'com.vanniktech.maven.publish'

--- a/lint-checks/src/test/java/com/uber/lintchecks/StringToCaseNoLocaleDetectorTest.kt
+++ b/lint-checks/src/test/java/com/uber/lintchecks/StringToCaseNoLocaleDetectorTest.kt
@@ -33,6 +33,11 @@ class StringToCaseNoLocaleDetectorTest : LintTestBase() {
             }
           }
         """))
+        // Ignored because Lint seems to run tests in two modes where `kotlin.uast.force.uinjectionhost` is
+        // on or off. The output is different in those two modes where the position at which the lin
+        // error underline happens is off by one. This doesn't mean that the bug is in the detector
+        // since it correctly identifies the issue.
+        .checkUInjectionHost(false)
         .detector(StringToCaseNoLocaleDetector())
         .issues(StringToCaseNoLocaleDetector.ISSUE)
         .run()

--- a/recipes/src/main/java/com/uber/lintchecks/recipes/guardrails/JavaOnlyDetector.kt
+++ b/recipes/src/main/java/com/uber/lintchecks/recipes/guardrails/JavaOnlyDetector.kt
@@ -170,8 +170,9 @@ class JavaOnlyDetector : Detector(), SourceCodeScanner {
       ): Pair<String, LintFix>? {
         return context.evaluator.getSuperMethod(node)?.let { method ->
           context.evaluator.findAnnotation(method, targetAnnotation)?.run {
+            val containingClass = checkNotNull(method.containingClass)
             val message = "Function overrides ${method.name} in ${UastLintUtils.getClassName(
-                method.containingClass)} which is annotated @$targetAnnotationSimpleName, it should also be annotated."
+                    containingClass)} which is annotated @$targetAnnotationSimpleName, it should also be annotated."
             val modifier = node.modifierList.children.joinToString(separator = " ") { it.text }
             return@let message to fix()
                 .replace()

--- a/sample/build.gradle
+++ b/sample/build.gradle
@@ -19,7 +19,6 @@ apply plugin: 'kotlin-android'
 
 android {
     compileSdkVersion deps.build.compileSdkVersion
-    buildToolsVersion deps.build.buildToolsVersion
 
     defaultConfig {
         applicationId "com.uber.myapplication"

--- a/sample/gradle.properties
+++ b/sample/gradle.properties
@@ -13,7 +13,4 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 #
-
-POM_NAME=YOUR PROJECTS NAME
-POM_ARTIFACT_ID=YOUR-ARTIFACT-ID
-POM_PACKAGING=android
+android.useAndroidX=true

